### PR TITLE
add server-side function distinct

### LINKED ISSUE #98: Support ERDDAP "distinct()" server-side function
erddapy doesn't currently support ERDDAP "Server-side functions". The only such functions I'm aware of are `distinct()` and various ordering functions. The latter don't really seem necessary in the context of erddapy and pandas, b/c sorting can be done client-side via Pandas. But `distinct()` does make a difference, as it can make the response much smaller, depending on the data.

It's actually very easy to use distinct with erddapy as-is. Once I have the request ready to go, I pass this url to pandas read_csv:
```python
e.get_download_url() + '&distinct()'
```

But figuring out how to do this takes a fair bit of work. It'd save users some effort if this was built into erddapy. I could be persuaded to try to add it myself and submit a PR, but not this month ...

### PR DESCRIPTION
**Pull request by ocefpaf**
Closes #98 

Work in progress, needs:

- [ ] docstrings
- [ ] notebook example
- [ ] expand the functionality too all returns

@emiliom this implements `distinct` only for `download_url` but it seems that it makes sense to add this too all URLs. Not sure what is the most sane way to do that yet.

PS: do you have a nice example of this in action for the notebook/documentation?

### CODE REVIEW

File: erddapy/erddapy.py [hunk 1]
```diff
@@ -22,6 +22,14 @@
 )
 
 
+def _distinct(url: str, **kwargs) -> str:
+    """Apply a server side function."""
+    distinct = kwargs.get("distinct")
+    if distinct:
```
Comment (emiliom) [inline erddapy/erddapy.py]: For what it's worth, this is testing for `distinct` having any value that produces a True in the if test, not just the value `True`. So, `distinct='abc'` will also result in the test being True.

**Comment by emiliom** [comment]
Looks good, other than my minor comment.

I haven't explored other urls other than `download_url`.

File: erddapy/erddapy.py (hunk 1 above)
Comment (ocefpaf) [reply erddapy/erddapy.py]: Yep. Still factoring it out in its own module so we can use typing and make it strict Boolean.

**Comment by emiliom** [comment]
Looking good!

**Pull request merged**